src/ledger/LedgerTxnAccountSQL.cpp

```diff
@@ -367,6 +370,348 @@ LedgerTxnRoot::Impl::deleteAccount(LedgerKey const& key)
     }
 }
 
+static void
+sociGenericBulkUpsertAccounts(Database& DB,
+                              std::vector<std::string> const& accountIDs,
+                              std::vector<int64_t> const& balances,
+                              std::vector<int64_t> const& seqNums,
+                              std::vector<uint32_t> const& subEntryNums,
+                              std::vector<std::string> const& inflationDests,
+                              std::vector<soci::indicator>& inflationDestInds,
+                              std::vector<uint32_t> const& flags,
+                              std::vector<std::string> const& homeDomains,
+                              std::vector<std::string> const& thresholds,
+                              std::vector<uint32_t> const& lastModifieds,
+                              std::vector<int64_t> const& buyingLiabilities,
+                              std::vector<int64_t> const& sellingLiabilities,
+                              std::vector<soci::indicator>& liabilitiesInds)
+{
+
+    std::string sql =
+        "INSERT INTO accounts ( "
```

Comment:
is there a difference between doing this and doing it the other way: `UPDATE ... ON CONFLICT ... INSERT` ?

Reply:
No idea. I'm not sure it actually works yet, still testing. Will try both.

Reply:
Turns out only the first exists; the second is not valid SQL. Because it's the world's most consistent language.